acapy_plugin_pickup/protocol/delivery.py

```diff
@@ -153,76 +153,104 @@ async def remove_message_for_key(self, key: str, msg: OutboundMessage):
         """Remove specified message from queue for key."""
 
 
-class InMemoryQueue(UndeliveredInterface):
-    """In-memory undelivered message queue."""
```

Comment:
The `InMemoryQueue` should not have been deleted in favor of the `PersistedQueue`. Both can coexist and the appropriate implementation for the use case should be selected by configuration.